airflow-core/src/airflow/api_fastapi/execution_api/datamodels/taskinstance.py

```diff
@@ -247,7 +247,7 @@ class TaskInstance(BaseModel):
     dag_id: str
     run_id: str
     try_number: int
-    dag_version_id: uuid.UUID
+    dag_version_id: uuid.UUID | None = None
```

Comment:
```suggestion
    dag_version_id: uuid.UUID | None
```

I feel we don’t really need the default, but can just explicitly provide None when needed. Also it’d be a good idea to document somewhere (a comment?) when None is expected.

Reply:
The TI.dag_version_id is nullable. We only enforce TI.dag_version_id on the constructor which can still create a TI even when the value of dag_version_id is null. So I think we don't need comment for this. However, let me know again if we should and I will add it